CMakeLists.txt

```diff
@@ -33,3 +33,12 @@ else ()
         include(cmake/config.cmake)
 endif()
 
+set(VERSION "0.3.0")
+set(GIT_HASH "")
+get_git_hash(GIT_HASH)
+message("Git hash is ${GIT_HASH}")
+
+configure_file(
+        ${PROJECT_SOURCE_DIR}/include/ylt/ylt.hpp.in
+        ${PROJECT_SOURCE_DIR}/include/ylt/ylt.hpp
```

Comment:
It might be better to put the genereated files in the build folder.

Reply:
Any example for generated file?

Reply:
https://cmake.org/cmake/help/latest/variable/PROJECT_BINARY_DIR.html#variable:PROJECT_BINARY_DIR

Reply:
> use `PROJECT_BINARY_DIR` and adjust include path too
> 
> ```
> configure_file(
>         ${PROJECT_SOURCE_DIR}/include/ylt/ylt.hpp.in
>         ${PROJECT_BINARY_DIR}/include/ylt/ylt.hpp
> ```
Put it in binray dir will make users confused since they can not fond any clues about generated files.
Actullay it's SOURCE FILES definitely.